File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/FillTextCommand_ViewTests.cpp
#include <gtest/gtest.h>
#include <memory>
#include "FillTextCommand_ViewModelTestEnvironment.h"
#include <string>
#include "FillTextCommand_ViewModel.h"
#include "FillTextCommand_ViewModelTestEnvironmentImpl.h"

namespace commands
{
  class FillTextCommand_ViewTests : public testing::Test
  {
  public:
    std::shared_ptr<FillTextCommand_ViewModel> sut;
    std::shared_ptr<FillTextCommand_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    virtual void when_fill_Changed_in_MyValue();
    virtual void when_fill_Changed_in_MyFreeValue();
  protected:
    void SetUp() override;
  };
  void FillTextCommand_ViewTests::SetUp()
  {
    this->testEnvironment = std::make_shared<FillTextCommand_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  TEST_F(FillTextCommand_ViewTests, Fill_Text_TextBox_given_when_fill_Changed_in_MyValue_then_)
  {
    this->BuildSut();
    this->when_fill_Changed_in_MyValue();
  }
  TEST_F(FillTextCommand_ViewTests, Fill_Text_ComboBox_given_when_fill_Changed_in_MyFreeValue_then_)
  {
    this->BuildSut();
    this->when_fill_Changed_in_MyFreeValue();
  }
  void FillTextCommand_ViewTests::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void FillTextCommand_ViewTests::when_fill_Changed_in_MyValue()
  {
    this->sut->myValueTextFilled(std::string("Changed"));
  }
  void FillTextCommand_ViewTests::when_fill_Changed_in_MyFreeValue()
  {
    this->sut->myFreeValueTextFilled(std::string("Changed"));
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/FillTextCommand_ViewTests.cs
/// <filename>
///     FillTextCommand_ViewTests.cs
/// </filename>
using commands;
using Microsoft.VisualStudio.TestTools.UnitTesting;

[ Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class FillTextCommand_ViewTests
{
    private commands.FillTextCommand_ViewModel sut;
 
    private commands.FillTextCommand_ViewModelTestSetup testSetup;
 
  [ Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
  public void SetUp()
  {
    this.testSetup = new commands.FillTextCommand_ViewModelTestSetupImpl();
    this.testSetup.Init();
  }
 
  [ Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
  public void Fill_Text_TextBox_given_when_fill_Changed_in_MyValue_then_()
  {
    this.BuildSut();
    this.when_fill_Changed_in_MyValue();
  }
 
  [ Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
  public void Fill_Text_ComboBox_given_when_fill_Changed_in_MyFreeValue_then_()
  {
    this.BuildSut();
    this.when_fill_Changed_in_MyFreeValue();
  }
 
    protected   virtual void BuildSut()
  {
    this.sut = this.testSetup.BuildSut();
  }
 
    public   virtual void when_fill_Changed_in_MyValue()
  {
    this.sut.myValueTextFilled("Changed");
  }
 
    public   virtual void when_fill_Changed_in_MyFreeValue()
  {
    this.sut.myFreeValueTextFilled("Changed");
  }
}
